regression/contracts/variant_parsing_multiple_clauses_fail/test.desc

```diff
@@ -0,0 +1,16 @@
+CORE
+main.c
+--apply-loop-contracts
+activate-multi-line-match
+^main.c.*error: syntax error before '__CPROVER_decreases'\n.*__CPROVER_decreases\(42\)$
+^PARSING ERROR$
+^EXIT=(1|64)$
+^SIGNAL=0$
+--
+--
+This test fails because we have multiple decreases clauses. CBMC only admits
+one decreases clause for each loop. If one wants to write a multidimensional
+decreases clause, it should be placed inside a single __CPROVER_decreases(...),
+where each component of the multidimensional decreases clause is separated by a
+comma. Hence, in this test, the correct syntax is 
+__CPROVER_decreases(N - i, 42).
```

Comment:
Nice! :) These test.desc files are very well written.

Reply:
Shouldn't we support multiple decreases clauses in the future?

Reply:
It's unclear as of now.

`__CPROVER_decreases` is unlike the other contracts. You can arbitrarily reorder multiple ENSURES, REQUIRES, ASSIGNS, LOOP_INVARIANT clauses, but the order of expressions matters for DECREASES clauses. They are not a conjunction (or a set), but an ordered tuple.

Moreover,

```
__CPROVER_decreases(X)
__CPROVER_decreases(Y)
```

reads as "both" X and Y monotonically decrease at each iteration, but for termination we simply need either one to monotonically decrease, and this sort of annotation would just be confusing.